knative-operator/pkg/controller/knativeserving/kourier/kourier.go

```diff
@@ -32,14 +32,6 @@ func Apply(instance *servingv1alpha1.KnativeServing, api client.Client, scheme *
 		return err
 	}
 
-	if instance.Status.IsFullySupported() {
-		// TODO: verify deployed kourier is not different from kourier-latest.yaml accurately.
-		if err := checkDeployments(&manifest, instance, api); err == nil {
-			return nil
-		}
-	}
-
-	// Us reaching here means we need to do something and/or wait longer.
 	instance.Status.MarkDependencyInstalling("Kourier")
```

Comment:
We mustn't do this without a condition I think. I think manifestival needs a `Check` function or something that allows us to check if we even need to apply an update or not. We only want to move into `Installing` state if we know we actually need to apply things.

Reply:
I see. In that case, we also should fix Serving manifest's `ApplyAll()`, I think.

https://github.com/openshift-knative/serving-operator/blob/331df80ba3d1aba7f74065b02bc982d4ae99f905/pkg/reconciler/knativeserving/knativeserving_controller.go#L197

Reply:
Never mind. I misunderstood the point.

Manifestival's `ApplyAll()` already have the function (`UpdateChanged()`) to decide if apply or not. But we only want to move into `Installing` state only when we need to apply things.

Reply:
Maybe for now, we should just stay in Installed=True but apply the updates anyway?

Reply:
Discussed this with @jcrossley3 in Slack.

Shall we try with the following pseudo-code flow?

```
func Apply(instance *servingv1alpha1.KnativeServing, api client.Client, scheme *runtime.Scheme) error {
  manifest := NewManifest(...)
  // apply transforms
  manifest.Apply()

  if err := checkDeployments(...); err != nil {
    instance.Status.MarkDependencyInstalling("Kourier")
  }
  instance.Status.MarkDependenciesInstalled()
  return api.Status().Update(context.TODO(), instance)
}
```

Is there any downside to this? 🤔

Reply:
did you mean _else_ `MarkDepsInstalled()`?

Reply:
yes I did, edited!